doc/reference/trd-hil-design.md

```diff
@@ -720,6 +723,42 @@ blocking APIs, not to never implement them.  They can and should at
 times exist, but their uses cases should be narrow and constrained as
 they are fundamentally not as reusable.
 
+Rule 11: Do not include `initialize()`
+===============================
+
+Avoid `initialize()` or `start()` or `on()` functions in HIL interfaces. While
+these types of methods are intuitive when implementing a HIL, they are often not
+intuitive for users of that HIL, as it is not often clear when `initialize()`
+should be called. For example, should `initialize()` be called once when the
+board boots? Or before every use? If there are multiple users of the peripheral
+(after a virtualization layer for example) then it isn't clear which user should
+call `initialize()`. In general, it is better to require that the implementation
+track the state of the underlying hardware and configure it as needed.
+
+See https://github.com/tock/tock/issues/1035 for more discussion.
+
+Rule 12: Do include `set_client()`
+===============================
+
+Include the `set_client()` function in the HIL if the HIL uses callbacks. Not
+all HILs require split-phase operation and need a client interface, but for
+those that do the `set_client()` function should be included in the HIL. This
+makes writing generic components for boards possible.
+
```

Comment:
This one seems pretty obvious today, but that's just to us. So yeah, this should definitely be in.